src/NServiceBus.Metrics/MetricsConfigurationExtensions.cs

```diff
@@ -19,8 +19,14 @@ public static MetricsOptions EnableMetrics(this EndpointConfiguration endpointCo
             Guard.AgainstNull(nameof(endpointConfiguration), endpointConfiguration);
 
             var settings = endpointConfiguration.GetSettings();
+            var options = settings.GetOrCreate<MetricsOptions>();
+
             settings.EnableFeatureByDefault<MetricsFeature>();
-            return settings.GetOrCreate<MetricsOptions>();
+
+            endpointConfiguration.Recoverability().Immediate(c => c.OnMessageBeingRetried((m, ct) => options.Immediate(m, ct)));
```

Comment:
@bording Is there a benefit to having the recoverability configured here in the EnableMetrics() extension method?  I am a novice at designing these Feature implementations, but I was thinking this would be in the Feature setup?  Always open to learning better ways, though.  Thanks!

Reply:
@bbrandt This is a bit of a weird case. Overall, it does make sense for a feature to set up everything it needs in its `Setup` method.

The problem here is that metrics needs to hook into recoverability notifications, but there isn't a nice way to do that because we've made the `RecoverabilitySettings` constructor internal. 

That's why Daniel had to resort to using `Activator.CreateInstance` when he moved it into the feature setup.

I moved it back to how it was originally because I felt that was the better overall tradeoff vs. resorting to a backdoor way to call a constructor we didn't make public.

Reply:
Got it!  Thanks for taking the time to explain.

Reply:
That being said the constructor doesn't have to be internal. We could change that since we are working on v9 currently. But in the grand scheme of things since in a send-only endpoint setting up recoverability has no effect we are not gaining much here to do it in the feature startup since that extra delegate allocation will anyway never be called.